docs: add a note on when not to use renderHook (#1093)
testing-library/testing-library-docs: docs/react-testing-library/api.mdx

This is a convenience wrapper around `render` with a custom test component. The
API emerged from a popular testing pattern and is mostly interesting for
libraries publishing hooks. 

> Do not use `renderHook` when your hook is easy to test by just testing the components using it.

Reviewer: ```suggestion
Do not use `renderHook` when your hook can be easily tested by testing the components using it.
```

Reviewer: Thanks @zaicevas!
A minor suggestion here.

> Alternatively, you should prefer `render` since a custom test

Reviewer: Is "alternatively" really needed here? I think we can remove it :)
